src/libtriton/format/pe/pe.cpp

```diff
@@ -50,62 +50,69 @@ namespace triton {
+      bool Pe::parse(void) {
         this->header.parse(this->raw, this->totalSize);
         return true;
       }
 
 
-      void PE::initMemoryMapping(void) {
+      void Pe::initMemoryMapping(void) {
         for (auto section : header.getSectionHeaders()) {
           triton::format::MemoryMapping area(this->raw);
 
-          if (this->totalSize < (section.rawAddress + section.rawSize)) {
-            std::cerr << "Warning PE::initMemoryMapping(): Some PE Section Headers of the binary file are corrupted." << std::endl;
+          triton::uint32 rawAddr = section.getRawAddress();
+          triton::uint32 rawSize = section.getRawSize();
+          triton::uint32 virtAddr = section.getVirtualAddress();
+          if (this->totalSize < (rawAddr + rawSize)) {
+            std::cerr << "Warning Pe::initMemoryMapping(): Some PE Section Headers of the binary file are corrupted." << std::endl;
             continue;
           }
 
-          area.setOffset(section.rawAddress);
-          area.setSize(section.rawSize);
-          area.setVirtualAddress(section.virtualAddress);
+          area.setOffset(rawAddr);
+          area.setSize(rawSize);
+          area.setVirtualAddress(virtAddr);
 
           this->memoryMapping.push_back(area);
         }
       }
 
 
-      triton::uint64 PE::getOffsetFromAddress(triton::uint64 vaddr) const {
+      triton::uint64 Pe::getOffsetFromAddress(triton::uint64 vaddr) const {
         for (auto section : this->header.getSectionHeaders()) {
-          if (vaddr >= section.virtualAddress && vaddr < (section.virtualAddress + section.rawSize)) {
-              return ((vaddr - section.virtualAddress) + section.rawAddress);
+          if (vaddr >= section.getVirtualAddress() && vaddr < (section.getVirtualAddress() + section.getRawSize())) {
+              return ((vaddr - section.getVirtualAddress()) + section.getRawAddress());
           }
         }
         return 0;
       }
 
-      void PE::initExportTable(void) {
-          triton::uint32 exportStart = header.getDataDirectory().exportTable_rva;
-          triton::uint32 exportSize = header.getDataDirectory().exportTable_size;
-          if (0 == exportStart) {
+      void Pe::initExportTable(void) {
+          triton::uint32 exportStart = header.getDataDirectory().getExportTable_rva();
+          triton::uint32 exportSize = header.getDataDirectory().getExportTable_size();
+
+          if (exportStart == 0) {
               std::memset(&exportTable, '\0', 40);
               return;
           }
-          std::memcpy(&exportTable, raw+getOffsetFromAddress(exportStart), 40);
-          exportTable.name = std::string((char*)raw+getOffsetFromAddress(exportTable.nameRVA));
-          triton::uint32 addrTableStart = getOffsetFromAddress(exportTable.exportAddressTableRVA);
-          for (triton::usize i=0; i<exportTable.addressTableEntries; ++i) {
-              PEExportEntry entry;
+
+          exportTable.parse(raw+getOffsetFromAddress(exportStart));
```

Comment:
Is it enough if I change getOffsetFromAddress to throw an exception if it is passed an address that doesn't map to a valid offset?